More button stuff, new dodgeballs, new spraypaint, kleiner revision

PYROTEKNIK opened the pull request:
Current Issues: 
Kleiners can sometimes become invincible, not sure of cause
Kleiner pathfinding could be more efficient

Please make sure you look over all this stuff i don't wanna break the server

Commit:
Update categories.lua

oops i forgot this shit

Commit:
Update init.lua

Fart fix

Commit:
fixed anims and sounds on monke swep

Commit:
fixed anims and sounds on monke swep

Commit:
Update weapon_monke.lua

Networking handling for reload function

Commit:
Update weapon_monke.lua

Networking handling for reload function

Commit:
Update weapon_monke.lua

Improved pony support

Commit:
Update weapon_monke.lua

Improved pony support

Commit:
Update weapon_monke.lua

Whoops forgot some testing stuff in there

Commit:
Update weapon_monke.lua

Whoops forgot some testing stuff in there

Commit:
Update weapon_monke.lua

random healing amount for fun :)

Commit:
Update weapon_monke.lua

Commit:
Update weapon_monke.lua

PYROTEKNIK committed:
Update weapon_monke.lua

networking adjustments

PYROTEKNIK committed:
Merge branch 'master' of https://github.com/PYROTEKNIK/contrib

PYROTEKNIK committed:
Nav Mesh, Random Outcome Buttons, Nextbot-Based Kleiners, Big Chungus in the bathroom

Known issues:
Kleiners can become invincible for some reason
Kleiner AI might need some further optimization
Other things to look at:
Button functions

PYROTEKNIK committed:
Update magicbutton.lua

cleaned up some stuff

PYROTEKNIK committed:
button model

oops i forgot the button model

PYROTEKNIK commented:
oh yeah you might want to tweak that MoneyPrize function in the magic buttons to something else if the prize value is too high

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,142 @@
+ 
+
+local spkr = Material("voice/icntlk_pl")
+hook.Add("CreateClientsideRagdoll","KleinerRagdollsFix",function(ent,rag)
```
Does the ragdoll color not work otherwise?

PYROTEKNIK replied:
GetPlayerColor only exists on player entity, so the material proxy typically only returns the default blue color unless you add that function to an entity.

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,142 @@
+		
+		
+		if(IsValid(rag))then rag:SetSubMaterial(5,ent:GetSubMaterial(5,"")) end
+		timer.Create(ent:EntIndex().."ragcleaner",10,1,function()
```
timer.Simple

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,142 @@
+end
+
+
+function ENT:Think()
```
Can this go into a draw hook, so it's not run all the time? Even if you have to call SetupBones

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,142 @@
+
+function ENT:GetPlayerColor()
+	if(self:GetBased())then
+	return Vector(0.8,0,0.05)
```
can these colors (since they are copied in the ragdoll) be put in a single location?

swampservers commented on lua/autorun/client/big_chungus_in_the_bathroom.lua:
```diff
@@ -0,0 +1,34 @@
+BIG_CHUNGUS_LOCATION = Vector(0, 1598, 64)	
```
add the header

swampservers replied:
add the header

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+		if(IsValid(attacker) and attacker:IsPlayer())then 
+		self:AddTargetViolence(attacker)
+			local reward = self:GetBased() and self.KillRewardBased or self.KillReward
+			if(attacker.PS_AddPoints)then attacker:PS_AddPoints(reward) end
```
PS_GivePoints

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+	if(ent.InVehicle and ent:InVehicle())then return false end
+	if(ent.Alive and !ent:Alive())then return false end
+	if(ent:GetMoveType() == MOVETYPE_FLY)then return false end
+	if(player.IsAFK and player:IsAFK())then return false end
```
mixing `ent` and `player`. Do you expect this to target entities other than players? If not, don't add extra checks

PYROTEKNIK replied:
they were designed originally to target eachother, but i figured it would be a waste of serverside logic to have them behave away from players. i will remove them

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+	return self:FindTarget()
+end
+
+function ENT:GetTargetTable()
```
Where is this used? Is there a reason you can't just use `KLEINER_NPC_TARGETS or {}` ?

PYROTEKNIK replied:
i think i was originally using this in place of KLEINER_NPC_TARGETS before i had that table. i will remove this

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+	end
+
+	if(playersum == 0)then --slowly die if they end up somewhere where nobody is
+		local dmg = DamageInfo()
```
SetHealth?

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+end
+
+
+function ENT:MoveToPos( pos, options )
```
wait, isn't this method built into the NextBot?

PYROTEKNIK replied:
i had to make an alternate implementation of it that runs  self:WhilePathing(), which allows movement to be interrupted for handling things like doors

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+end
+
+function ENT:GetNextPathPoint(ahead)
+if(!self.path or !self.path:IsValid())then return end
```
can you run a formatter please

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+	end
+	
+	--NOTE: door use seems to be pretty expensive due to the tracing, i would like to optimize this so that it doesn't run unless the path contains a doorway.
+	--Disabling the following function will just make them get stuck and teleport past the door instead
```
teleporting is fine

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,970 @@
+		end
+		
+		--if(!self:PosInRange(area:GetCenter()))then return -1 end
+		// compute distance traveled along path so far
```
please do not use // comments in glua, it messes up the formatting in just about every editor

PYROTEKNIK replied:
whoops that's leftover from the original function example on the wiki

swampservers commented on lua/entities/kleiner/sv_spawning.lua:
```diff
@@ -0,0 +1,62 @@
+		KLEINER_NPCS[ent] = ent:GetClass() == "kleiner" and true or nil
+		KLEINER_NPC_TARGETS[ent] =  ent:IsPlayer() and true or nil
+		
+		KLEINER_NPCS_FILTER = {}
```
Remove this, use a function for the trace filter instead (if you were going to use it)

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,457 @@
+				table.insert(origins,trace)
+				local ext = area:GetExtentInfo()
+				local bnds = Vector(ext.SizeX,ext.SizeY,ext.SizeZ + 1)
+				debugoverlay.Box( area:GetCenter() + Vector(0,0,0.5) , bnds*-0.5, bnds*0.5, 60, Color( 0, 0, 255,0 ) )
```
remove

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,457 @@
+
+function MoneyPrize() -- this is where we generate a random amount of points to give the player
+	local basenumber = 100
+	for i=1,math.random(10,40) do
```
Dude, what???

If you want a non-linear sampling probability (eg, more likely to choose a small number) do something like 

`math.pow(math.random(), 5) * (MAX-MIN) + MIN`

PYROTEKNIK replied:
i'm not good with that kind of math lol

swampservers replied:
the secret is that math.random() is 0 to 1, and pow(math.random(), anynumber) is also 0 to 1. if anynumber>1, the result biases more toward 0, if anynumber < 1 the result is biased more toward 1

PYROTEKNIK replied:
for some reason i thought pow() was the same as number^x

swampservers replied:
pow(a,b) is a^b

PYROTEKNIK replied:
oh right yeah i guess it makes sense that using values between 0 and 1 always returns values between 0 and 1.  usually when i use it i would use values above 1

swampservers commented:
Can you remove the nav file? I'll install it

PYROTEKNIK committed:
Requested Fixes, Kleiner Larvae SWEP

PYROTEKNIK committed:
Removed NAV mesh

PYROTEKNIK committed:
haha whoops

PYROTEKNIK committed:
tweaked bait drop chance

PYROTEKNIK committed:
Lose health while stuck

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,154 @@
+end
+
+
+function ENT:Think()
```
clean up the empty function

PYROTEKNIK committed:
Update big_chungus_in_the_bathroom.lua

header

PYROTEKNIK committed:
Update cl_init.lua

PYROTEKNIK committed:
Update weapon_monke.lua

i don't even know what's going on with this anymore. my file is the best one i promise

PYROTEKNIK committed:
Update weapon_monke.lua

formatting

PYROTEKNIK committed:
Update weapon_monke.lua

removed conflicts, change to rendering

PYROTEKNIK committed:
Update weapon_monke.lua

PYROTEKNIK committed:
Merge branch 'master' into master

swampservers commented:
@PYROTEKNIK Bro its still not formatted

PYROTEKNIK commented:
i ran it on that file. do you need me to format everything else?

swampservers commented:
Every file... why would I give you directions to batch format the entire folder if i only wanted one file?

PYROTEKNIK commented:
you told me you converted everything already

PYROTEKNIK committed:
Merge branch 'master' of https://github.com/PYROTEKNIK/contrib

swampservers commented:
No, I formatted everything that was in contrib already. I can't format the files in your PR because they are in your fork which I can't edit.

PYROTEKNIK committed:
formatting

swampservers commented on format.bat:
```diff
@@ -0,0 +1,3 @@
+@echo off
```
I don't want this file

swampservers commented on lua/autorun/client/big_chungus_in_the_bathroom.lua:
```diff
@@ -0,0 +1,34 @@
+end
+
+timer.Create("Chungus_Checker", 1, 0, function()
+    if (true or LocalPlayer().GetLocationName and LocalPlayer():GetLocationName() == "Bathroom") then
```
`if true or`... ?

swampservers replied:
GetLocationName shouldn't need to be checked for anymore, since it is in contrib

PYROTEKNIK replied:
sorry, i can't test this stuff in cinema so i occasionally forget to remove these checks

swampservers replied:
Shouldn't need the checks anymore for location-related stuff

PYROTEKNIK committed:
removed extra file

PYROTEKNIK committed:
cleaned up debug code

PYROTEKNIK committed:
Kleiner Spawning can be changed via ConVar

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,139 @@
+﻿-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+-- INSTALL: CINEMA
+if (SERVER) then return end
```
this never happens

swampservers commented on lua/entities/kleiner/cl_init.lua:
```diff
@@ -0,0 +1,139 @@
+	return  math.max(math.floor( (2.8^star)),1)
+end
+
+local function KleinerViolenceStars(value)
```
dude what if it was like GTA where you kill the kleiners and then they send like armored kleiners against you lmao, and it showed 1-5 stars in the corner of the screen

PYROTEKNIK replied:
i could do something like that if you're interested. I just wasn't sure if i wanted to include the current hud since it looked kind of shitty. I'm not sure about armor since kleiners are meant to interact with other players too. I can look into this if you'd like, though.

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,916 @@
+        self:AddGestureSequence(self:LookupSequence("fear_reaction_gesture"))
+    end
+
+    if (self:Health() < 0 and self.IsAlive ~= true) then
```
`and not self.IsAlive` ?

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,916 @@
+end
+
+function ENT:DropGrenades()
+    if (IsValid(self.Grenade1)) then
```
You could do `for _,gren in ipairs({self.Grenade1,self.Grenade2}) do` (not mandatory, but that is good practice)

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,916 @@
+    end
+end
+
+if (SERVER) then
```
unnecessary check

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,916 @@
+    if (ent.Alive and not ent:Alive()) then return false end
+    if (ent:GetMoveType() == MOVETYPE_FLY) then return false end
+    if (ent.IsAFK and ent:IsAFK()) then return false end
+    if (Safe and Safe(ent) and ent:IsPlayer()) then return false end
```
Is `ent` ever not expected to be a player?

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,916 @@
+                        self:ResetBehavior()
+                    end
+
+                    --cheap method of keeping kleiners with similar targets from clumping together. i figure it's cheaper than some kind of avoidance.
```
One thing that might work (it worked for me doing something similar in the past) is, if two kleiners intersect with each other, the one farther away from the target stops until the intersection ends

PYROTEKNIK replied:
wouldn't that require extra server work doing collisions with eachother

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,916 @@
+
+ENT.PathGen = function(area, fromArea, ladder, elevator, length)
+    KLPATHGEN_ITERS = (KLPATHGEN_ITERS or 0) + 1
+    local self = RELEVANT_KLEINER --this is bullshit, i guess this callback doesn't include the entity pathing. 
```
You could also make this a local function inside ENT:ChaseTarget to capture the local `self` variable

PYROTEKNIK committed:
Kleiner Door Handling

Kleiners will attempt to teleport if they touch a door while moving along a path. This should be slightly faster than waiting until they are identified as stuck.  ClearStuck may be delaying actual stuck handling

swampservers commented on lua/entities/kleiner/sv_spawning.lua:
```diff
@@ -0,0 +1,93 @@
+    KLEINER_DESIRED_NUMBER = GetConVar("kleiner_spawncount"):GetInt() or 0
+    timer.Destroy("kleiner_spawner")
+
+    if (value > 0) then
```
I think this check is unnecessary. Also don't see why you pass `value` to this function when it seems like its always the same value as `KLEINER_DESIRED_NUMBER` which you get inside the function anyway?

PYROTEKNIK replied:
value is passed from the convar's change callback to update it.  i should change these to actually use the passed value

swampservers commented on lua/entities/kleiner/sv_spawning.lua:
```diff
@@ -0,0 +1,93 @@
+    end
+
+    --if grenade is hit by high force damage from non-world, change grenade owner to attacker
+    if (target:GetClass() == "npc_grenade_frag" and dmginfo:GetDamage() > 5 and dmginfo:GetDamageForce():Length() > 200 and IsValid(dmginfo:GetAttacker()) and IsValid(dmginfo:GetInflictor())) then
```
is this so you can shoot a grenade and it explodes? uh, okay

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+
+ENT.PlacementSettings.SurfacePropBlacklist = {"dirt", "grass", "gravel", "rock"}
+
+--**studio** is any model
```
If models don't work, what about displacements? I think there is a special material for them too

PYROTEKNIK replied:
some displacements are not terrain. These checks are intended to prevent placing them on surfaces that don't make sense for a button to be, like floating on a disjointed part of a model's collision geometry, or on natural surfaces.

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+                table.insert(origins, trace)
+                local ext = area:GetExtentInfo()
+                local bnds = Vector(ext.SizeX, ext.SizeY, ext.SizeZ + 1)
+                debugoverlay.Box(area:GetCenter() + Vector(0, 0, 0.5), bnds * -0.5, bnds * 0.5, 60, Color(0, 0, 255, 0))
```
debug overlay?

PYROTEKNIK replied:
These are debugging functions used for visualizing. They apparently do nothing unless developer is set to 1.

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+        end
+
+        if (table.Count(navareas) > 0) then
+            debugoverlay.SweptBox(tr2.start, trace2.HitPos, tr2.mins, tr2.maxs, Angle(), 60, Color(0, 255, 0, 0))
```
overlay

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+    end
+
+    if (trace1 and trace2) then return trace2, trace1 end --print("SUCCESS AFTER "..TOTAL_FAILURES.." ATTEMPTS!")
+    --print("SUPER BIG FAILURE! HOW IS IT POSSIBLE?")
```
You'd probably want to use `Error()` or `assert()` in the future

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+    local lc = render.ComputeLighting(pos, self:GetUp())
+    local light1 = {}
+    light1.type = MATERIAL_LIGHT_POINT
+    light1.color = cvector * 15
```
So there is like a light on the button? what do you think about animating the intensity over time with `sin(SysTime())` ?

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+function MoneyPrize()
+    local basenumber = 100
+
+    for i = 1, math.random(10, 40) do
```
can you replace this with actual math please

PYROTEKNIK replied:
I swear i did that already. I think i lost that change.

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+    basenumber = math.Clamp(basenumber, 1000, math.random(100000, 500000))
+    basenumber = math.Round(basenumber, -3)
+
+    return basenumber * 1
```
*1

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+    local counter = 0
+
+    while (item == nil and counter < 100) do
+        effect = table.Copy(table.Random(MagicButtonOutcomes))
```
Don't see why you need to copy it

swampservers commented on lua/entities/magicbutton.lua:
```diff
@@ -0,0 +1,463 @@
+    local item
+    local counter = 0
+
+    while (item == nil and counter < 100) do
```
Maybe you can shuffle the order of `MagicButtonOutcomes`, then iterate it, and return early if one of the effects "works"?

swampservers commented on lua/autorun/client/big_chungus_in_the_bathroom.lua:
```diff
@@ -0,0 +1,38 @@
+BIG_CHUNGUS_LOCATION = Vector(0, 1598, 64)
+BIG_CHUNGUS_NORMAL = Vector(1, 0, 0)
+if (game.GetMap() ~= "cinema_swamp_v3") then return end
+if (engine.ActiveGamemode() ~= "cinema") then return end
```
remove check

swampservers commented:
This is looking good, I'll test it soon and see if there are any other issues!

PYROTEKNIK committed:
Removed unneeded checks.

PYROTEKNIK committed:
Proper use of passed value

PYROTEKNIK committed:
unneeded gamemode check

PYROTEKNIK committed:
unneeded check

PYROTEKNIK committed:
unnecessary checks

PYROTEKNIK committed:
unnecessary checks

PYROTEKNIK committed:
no error on weird lights that don't have a dietime

PYROTEKNIK committed:
Update magicbutton.lua

Outcome is chosen in a more efficient manner

swampservers commented on lua/entities/kleiner/sv_spawning.lua:
```diff
@@ -0,0 +1,93 @@
+                local newkleiner = ents.Create("kleiner")
+
+                if (IsValid(newkleiner)) then
+                    local spawnpoint = gmod.GetGamemode().SpawnPoints and table.Random(gmod.GetGamemode().SpawnPoints):GetPos() or Vector(0, 0, 16)
```
This line threw a NULL entity error

swampservers replied:
probably just ents.FindByClass("info_player_start") then cache it, and if you find a NULL entity, update the cache

swampservers commented on lua/entities/kleiner/sv_spawning.lua:
```diff
@@ -0,0 +1,93 @@
+KLEINER_NPCS_CURRENT_NUMBER = KLEINER_NPCS_CURRENT_NUMBER or 0
+KLEINER_DESIRED_NUMBER = KLEINER_DESIRED_NUMBER or GetConVar("kleiner_spawncount"):GetInt() or 0
+
+function KLEINER_NPC_SETUPTIMER(value)
```
this is overcomplicated and doesn't work when i change the convar. Just do `timer.Create` in the global scope (eg. not inside a function) and check the `GetConVar("kleiner_spawncount"):GetInt()` every time. There is no reason to use `hook.Add("Initialize"`; just put it in the global scope.

swampservers commented on lua/entities/kleiner/init.lua:
```diff
@@ -0,0 +1,911 @@
+                    wanderpos = self.WanderForcePos + VectorRand() * Vector(1, 1, 0):GetNormalized() * math.Rand(80, 150)
+                end
+
+                wanderpos = wanderpos or self:FindSpot("random", {
```
This seems to be the main lag causer, looking at https://github.com/Facepunch/garrysmod/blob/be251723824643351cb88a969818425d1ddf42b3/garrysmod/gamemodes/base/entities/entities/base_nextbot/sv_nextbot.lua it seems to essentially find a bunch of nearby areas, then make a path to each area (!!!). We should probalby use a much lower radius so it finds less areas, and it can find its way to distant areas after multiple wanders. It might also work to override ent:FindSpots to not calculate a path and just use the vector:Distance() instead.

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master' into master

PYROTEKNIK committed:
Kleiner Bugfixes

Bait are cleaned up properly
Voice filter and speaker icon are removed
Teleport effect is less jarring
Wander radius reduced
Removed unneeded normal map from joker texture

PYROTEKNIK committed:
Update magicbutton.lua

Checks for kleiners

PYROTEKNIK committed:
Update weapon_monke.lua

Uses sequence info for activity enum instead of _G

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master' into master

PYROTEKNIK committed:
Update sv_spawning.lua

debug removed

PYROTEKNIK committed:
Update weapon_monke.lua

error fix

PYROTEKNIK committed:
voice filter removed, bugfixes

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master'

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master'

PYROTEKNIK committed:
Monke Error fixes

PYROTEKNIK committed:
Combine helmet in pointshop

PYROTEKNIK committed:
New dodgeball

new model, better swep

PYROTEKNIK committed:
More button outcomes, button despawning

PYROTEKNIK committed:
Improved Spraypaint

New model, changeable colors, resizable spray

PYROTEKNIK committed:
Update weapon_monke.lua

formatting

PYROTEKNIK committed:
Kleiner bait rebalanced for shop

PYROTEKNIK committed:
Update sv_spawning.lua

Kleiner spawning tweaked. Kleiners are less likely to kill eachother with grenades.

PYROTEKNIK committed:
Kleiners Revision

-no longer make engine footstep sounds
-no longer damage other kleiners while alive
-no longer drop bait
-better wandering function
-capable of opening and avoiding doors

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master'

PYROTEKNIK committed:
Delete shared.lua

PYROTEKNIK committed:
removed temp func

PYROTEKNIK committed:
removed "ammo" from new spraypaint

PYROTEKNIK committed:
Update weapon_spraypaint.lua

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master'

PYROTEKNIK committed:
Update swamp_core.lua

oopsie

PYROTEKNIK committed:
Merge remote-tracking branch 'upstream/master'

swampservers commented on lua/autorun/swamp_core.lua:
```diff
@@ -57,7 +57,9 @@ function math.power2(n)
 end
 
--- TABLE
+BLACK = Color(0, 0, 0, 255)
```
there shouldn't be my changes in this PR, i think you need to fetch and merge upstream/master then push to origin/master

PYROTEKNIK commented:
This is where I had originally inserted vec:inbox

On Wed, May 5, 2021, 8:49 PM swampservers, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In lua/autorun/swamp_core.lua
> <https://github.com/swampservers/contrib/pull/223#discussion_r627058604>:
>
> > @@ -57,7 +57,9 @@ function math.power2(n)
>      return math.pow(2, math.ceil(math.log(n) / math.log(2)))
>  end
>
> --- TABLE
> +BLACK = Color(0, 0, 0, 255)
>
> there shouldn't be my changes in this PR, i think you need to fetch and
> merge upstream/master then push to origin/master
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/swampservers/contrib/pull/223#pullrequestreview-652936292>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA46FOLRJGBPLCMKPKUPTO3TMIGUXANCNFSM42FFIRXQ>
> .
>

PYROTEKNIK committed:
Updated neckbeard model

PYROTEKNIK committed:
Tweak to magicbutton chances, re-enabled object spawning outcome
